lib/active_set/sorting/active_record_strategy.rb

```diff
@@ -41,15 +41,24 @@ def executable_instructions
       private
 
       # https://stackoverflow.com/a/44912964/2884386
-      # Force null values to be sorted as if larger than any non-null value
+      # When ActiveSet.configuration.on_asc_sort_nils_come == :last
+      # null values to be sorted as if larger than any non-null value.
       # ASC => [-2, -1, 1, 2, nil]
       # DESC => [nil, 2, 1, -1, -2]
+      # Otherwise sort nulls as if smaller than any non-null value.
+      # ASC => [nil, -2, -1, 1, 2]
+      # DESC => [2, 1, -1, -2, nil]
       def order_operation_for(set_instruction)
         attribute_model = set_instruction.attribute_model
 
         arel_column = set_instruction.arel_column
         arel_direction = direction_operator(set_instruction.value)
-        nil_sorter = arel_column.send(arel_direction == :asc ? :eq : :not_eq, nil)
+
+        nil_sorter = if ActiveSet.configuration.on_asc_sort_nils_come == :last
```

Comment:
Given the conditional fork now, let's just move this into a method in this class:

```ruby
def nil_sorter_for(column, direction)
  nil_sorter_operator = if ActiveSet.configuration.on_asc_sort_nils_come == :last
                          direction == :asc ? :eq : :not_eq
                        else
                          direction == :asc ? :not_eq : :eq
                        end
  
  arel_column.send(nil_sorter_operator, nil)
end
```

I don't like the branching logic of that `nil_sorter_operator` variable, but I can't think of anything concise and equally as clear at the moment

Reply:
I agree with you that there feels like there must be a purer way of doing this. See c6a8b86 for the refactoring you suggested (which is an improvement at least).